### PR TITLE
Sortable

### PR DESCRIPTION
**Pull request by rcaught**
- Base logic for sanitizing, storing and calling sortables
- Helpers for creating links that show direction of sortables
- Allow pagination to pass-through sortables
- Custom placeholder text for search

### CODE REVIEW

File: spec/lib/lipstick/sortable_spec.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+module Lipstick
+  RSpec.describe Sortable do
+    context 'direction' do
+      it 'sanitizes' do
+        expect(Sortable.direction('hackerz')).to eq(
+          direction: 'asc', opposite: 'desc'
+        )
```
Comment (chrisjones120) [inline spec/lib/lipstick/sortable_spec.rb]: nice

File: lib/lipstick/sortable.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      attr_reader :sortable_by_cases, :sortable_by_else
+
+      def sortable_by(sort_cases, sort_else)
+        @sortable_by_cases = sort_cases
+        @sortable_by_else = sort_else
+      end
+
+      def sortable_sort(sort_by, direction)
+        sort_by = sort_by&.to_sym
+        direction = Lipstick::Sortable.direction(direction)[:direction]
```
Comment (chrisjones120) [inline lib/lipstick/sortable.rb]: I feel as though the variable / function naming here lacks imagination

File: lib/lipstick/sortable.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        directions = %w[asc desc]
+
+        direction = directions.find { |x| x == direction&.downcase }
+        direction ||= directions.first
+
+        {
+          direction: direction,
+          opposite: (directions - [direction]).first
+        }
+      end
```
Comment (chrisjones120) [inline lib/lipstick/sortable.rb]: I think the opposite could be done in a separate function - that way you don't need to return a hash & the function name would be more descriptive.

EG. get direction, then have a function called opposite that takes a direction and returns the opposite

**Comment by chrisjones120** [comment]
Very nice

**Comment by bradleybeddoes** [comment]
Keep this going @chrisjones120 don't let @rcaught get his dodgy code into prod.

**Pull request merged**